engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/stores/ExecutionStore.scala

```diff
@@ -24,22 +24,20 @@ object ExecutionStore {
       * false otherwise.
       */
     def allDependenciesAreIn(statusTable: Table[GraphNode, ExecutionIndex.ExecutionIndex, JobKey]) = {
-      def portIsInStatus(outputPort: OutputPort) = {
-        // If the port is a scatter gather port, then it depends on the collector node which is stored at index None
-        val index = outputPort match {
-          case _: ScatterGathererPort => None
-          case _ => key.index
-        }
-
-        outputPort.executionNode.isInStatus(index, statusTable)
+      def chooseIndex(port: OutputPort) = port match {
+        case _: ScatterGathererPort => None
+        case _ => key.index
       }
 
       key match {
         // For a scatter collector key, the dependencies are all the shards
-        case scatterCollector: ScatterCollectorKey => statusTable.row(scatterCollector.node).size == scatterCollector.scatterWidth
-        case conditionalCollector: ConditionalCollectorKey => statusTable.contains(conditionalCollector.node.executionNode, key.index)
+        case scatterCollector: ScatterCollectorKey =>
+          statusTable.row(scatterCollector.outputNodeToGather.singleInputPort.upstream.executionNode).size == scatterCollector.scatterWidth
```

Comment:
I think dropping a comment here as to why we have to go find the `scatterCollector.outputNodeToGather.singleInputPort.upstream.executionNode` would be valuable, same below for conditional.